src/webots/gui/WbGuiApplication.cpp

```diff
@@ -193,117 +194,206 @@ void WbGuiApplication::parseStreamArguments(const QString &streamArguments) {
+       "information is prepended into the log file"),
+    tr("file"));
+  QCommandLineOption convertOption("convert", tr("Convert a PROTO file to a URDF, WBO, or WRL file."), tr("file"));
+  parser.addOption(convertOption);
+  QCommandLineOption updateWorldOption("update-world", tr("Update world"));
+  parser.addOption(updateWorldOption);
+  QCommandLineOption x3DMetaFileExportOption("enable-x3d-meta-file-export", tr("Enable x3d meta file export"));
+  parser.addOption(x3DMetaFileExportOption);
+  parser.addOption(logPerfomanceOption);
+#ifndef _WIN32
+
+  parser.addOptions({
+    {"disable-logging", tr("Disable logging")},
+    {"enable-logging", tr("Enable logging")},
+    {"log-level", tr("Log level")},
+    {"no-sandbox", tr("No sandbox")},
+    {"single-process", tr("Single Process")},
+    {"remote-debugging-port", tr("Remote Debugging Port")},
+  });
+#endif
 
-  const int size = args.size();
-  for (int i = 1; i < size; ++i) {
-    const QString &arg = args[i];
-    if (arg == "--minimize")
+#ifdef _WIN32
+  if (qgetenv("WEBOTS_TERMINAL") == "1" && GetConsoleWindow() == 0)
+    AttachConsole(ATTACH_PARENT_PROCESS);
+#endif
+  parser.process(QCoreApplication::arguments());
+  QStringList args = parser.optionNames();
+  const QStringList path = parser.positionalArguments();
+
+  while (!args.isEmpty()) {
+    const QString subCommand = args.first();
+    if (subCommand == "help" && parser.isSet(helpOption)) {
+      args.pop_front();
+      mTask = HELP;
+    } else if (subCommand == "version" && parser.isSet(versionOption)) {
+      args.pop_front();
+      mTask = VERSION;
+    } else if (subCommand == "minimize" && parser.isSet(minimizeOption)) {
+      args.pop_front();
       mShouldMinimize = true;
-    else if (arg == "--fullscreen")
+    } else if (subCommand == "fullscreen" && parser.isSet(fullscreenOption)) {
+      args.pop_front();
       mShouldStartFullscreen = true;
-    else if (arg == "--mode=stop") {
-      cout << tr("The '--mode=stop' option is deprecated. Please use '--mode=pause' instead.").toUtf8().constData() << endl;
-      mStartupMode = WbSimulationState::PAUSE;
-    } else if (arg == "--mode=pause")
-      mStartupMode = WbSimulationState::PAUSE;
-    else if (arg == "--mode=realtime")
-      mStartupMode = WbSimulationState::REALTIME;
-    else if (arg == "--mode=fast")
-      mStartupMode = WbSimulationState::FAST;
-    else if (arg == "--mode=run") {
-      cout << "Warning: `run` mode is deprecated, falling back to `fast` mode" << endl;
-      mStartupMode = WbSimulationState::FAST;
-    } else if (arg == "--no-rendering")
+    } else if (subCommand == "mode" && parser.isSet(modeOption)) {
+      args.pop_front();
+      const QStringList mode = parser.values(modeOption);
+      if (mode.size() != 1 || mode[0] == "=") {
+        cerr << tr("Error: Must specify one mode argument.").toUtf8().constData() << endl;
+      }
```

Comment:
[coding style](https://github.com/cyberbotics/webots/wiki/CPP-Coding-Style#cs360-single-if-while-or-for-statement-must-be-written-without-brackets-the-statement-part-should-be-put-on-a-separate-line-in-case-of-nested-block-keep-brackets) (here and at other places):
```suggestion
      if (mode.size() != 1 || mode[0] == "=")
        cerr << tr("Error: Must specify one mode argument.").toUtf8().constData() << endl;
```